app/lib/internet_archive/archival.rb

```diff
@@ -0,0 +1,48 @@
+# Uses the Save Page Now 2 API
+# (https://docs.google.com/document/d/1Nsv52MvSjbLb2PCpHlat0gkzw0EvtSgpKHu4mk0MnrA)
+class InternetArchive::Archival
+  attr_reader :website_url, :job_id
+
+  def initialize(website_url)
+    @website_url = website_url
+  end
+
+  def request
+    response = connection.post("save", "url=#{website_url}").body
+    @job_id = response["job_id"]
+    response
+  end
+
+  def finished?
+    status["status"] == "success"
+  end
+
+  def status
+    connection.get("save/status/#{job_id}").body
```

Comment:
```suggestion
    connection.get("save/status/#{job_id}").body&.[]("status")
```
and refactor the other code so it doesn't say `status["status"]`